core/src/main/scala/org/bitcoins/core/protocol/ln/util/LnUtil.scala

```diff
@@ -40,42 +41,48 @@ trait LnUtil {
     encoded
   }
 
-  def decodeDataLength(u5s: Vector[UInt5]): Long = {
+  def decodeDataLength(u5s: List[UInt5]): Long = {
     require(u5s.length == 2,
             s"Data Length is required to be 10 bits, got ${u5s.length}")
-    decodeNumber(u5s)
+    decodeNumber(u5s).toLong
   }
 
   /** Returns a 5 bit bytevector with the encoded number for a LN invoice */
   @tailrec
   final def encodeNumber(
       len: BigInt,
-      accum: Vector[UInt5] = Vector.empty): Vector[UInt5] = {
+      accum: List[UInt5] = List.empty): List[UInt5] = {
     val quotient = len / 32
     val remainder = UInt5(len % 32)
     if (quotient >= 32) {
-      encodeNumber(quotient, remainder +: accum)
+      encodeNumber(quotient, remainder :: accum)
     } else if (quotient == 0) {
-      remainder +: accum
+      remainder :: accum
     } else {
       val quo = UInt5.fromByte(quotient.toByte)
-      val v = Vector(quo, remainder)
-      v ++ accum
+      quo :: remainder :: accum
     }
 
   }
 
+
+  /** Decodes a number from Bech32 to a long */
   @tailrec
-  final def decodeNumber(vector: Vector[UInt5], accum: Long = 0): Long = {
+  final def decodeNumber(list: List[UInt5], accum: BigInt = 0): BigInt = {
 
-    if (vector.isEmpty) accum
-    else if (vector.size == 1) {
-      decodeNumber(vector.tail, vector.head.toInt + accum)
-    } else {
-      val n = BigInt(32).pow(vector.size - 1)
-      val newAccum = vector.head.toBigInt * n + accum
-      decodeNumber(vector.tail, newAccum.toLong)
+    val b32 = BigInt(32)
```

Comment:
Should be moved back inside the ` h :: t` case otherwise it is pointless construction every time we decode